[BUG] Type checks failures

# Bug Description

Type check failures

## **Output of `OLS` version**

0.2.1

## **Describe the bug**

```
ols/src/prompts/prompt_generator.py:46: error: Unsupported operand types for + ("str" and "list[str | dict[Any, Any]]")  [operator]
ols/src/prompts/prompt_generator.py:46: note: Right operand is of type "str | list[str | dict[Any, Any]]"
ols/src/prompts/prompt_generator.py:48: error: Unsupported operand types for + ("str" and "list[str | dict[Any, Any]]")  [operator]
ols/src/prompts/prompt_generator.py:48: note: Right operand is of type "str | list[str | dict[Any, Any]]"
ols/src/prompts/prompt_generator.py:115: error: Unsupported operand types for + ("str" and "list[str | dict[Any, Any]]")  [operator]
ols/src/prompts/prompt_generator.py:115: note: Right operand is of type "str | list[str | dict[Any, Any]]"
ols/src/query_helpers/docs_summarizer.py:110: error: Incompatible types in assignment (expression has type "list[str]", variable has type "list[BaseMessage] | None")  [assignment]
```

--- a/ols/app/models/models.py
+++ b/ols/app/models/models.py
@@ -2,7 +2,7 @@
 
 import json
 from collections import OrderedDict
-from typing import Any, Dict, Optional, Self, Union
+from typing import Any, Optional, Self, Union
 
 from langchain.llms.base import LLM
 from langchain_core.messages import AIMessage, BaseMessage, HumanMessage
@@ -760,8 +760,8 @@ class MessageDecoder(json.JSONDecoder):
         super().__init__(object_hook=self._decode_message, *args, **kwargs)
 
     def _decode_message(
-        self, dct: Dict[str, Any]
-    ) -> Union[HumanMessage, AIMessage, Dict[str, Any]]:
+        self, dct: dict[str, Any]
+    ) -> Union[HumanMessage, AIMessage, dict[str, Any]]:
         """Decode JSON dictionary into Message objects if applicable.
 
         Args:
@@ -771,6 +771,8 @@ class MessageDecoder(json.JSONDecoder):
             Union[HumanMessage, AIMessage, dict]: A Message object if the input
             dictionary represents a message, otherwise returns the original dictionary.
         """
+        message: BaseMessage
+
         if "type" in dct:
             if dct["type"] == "human":
                 message = HumanMessage(content=dct["content"])

